src/core/src/MatrixDynSize.cpp

```diff
@@ -78,33 +78,35 @@ MatrixDynSize::MatrixDynSize(const MatrixDynSize& other)
 
 MatrixDynSize& MatrixDynSize::operator=(const MatrixDynSize& other)
 {
-    if (this == &other) return *this;
-
-    // Copy the new rows and columns
-    m_rows = other.m_rows;
-    m_cols = other.m_cols;
-    unsigned requiredCapacity = m_rows * m_cols;
-
-    // if other is empty, return
-    if (requiredCapacity == 0) return  *this;
-
-    // If the copied data fits in the currently allocated buffer,
-    // use that one (if the user want to free the memory can use
-    // the shrink_to_fit method).
-    // Otherwise, allocate a new buffer after deleting the old one)
-    if (m_capacity < requiredCapacity) {
-        // need to allocate new buffer
-        // if old buffer exists, delete it
-        if (m_capacity > 0) {
-            delete [] m_data;
-        }
-        m_data = new double[requiredCapacity];
-        m_capacity = requiredCapacity;
-    }
-
-    // Now we are sure that m_capacity >= m_rows*m_cols
-    std::memcpy(this->m_data, other.m_data, requiredCapacity * sizeof(double));
-    return *this;
+    return this->operator=(other);
```

Comment:
Can we do the explicit conversion here? Otherwise the piece of code is quite enigmatic, it seems as if teh method was calling itself recursively.